cli/pa_analyze_command.go

```diff
@@ -0,0 +1,980 @@
+			accountsWithStreams[accountName] = nil
+
+			streamTag := archive.TagStream(streamName)
+			serverNames := r.GetStreamServerNames(accountName, streamName)
+
+			cmd.logDebug(
+				"Inspecting account '%s' stream '%s', found %d servers: %v",
+				accountName,
+				streamName,
+				len(serverNames),
+				serverNames,
+			)
+
+			// Create map server->streamDetails
+			replicasStreamDetails := make(map[string]*server.StreamDetail, len(serverNames))
+			streamIsEmpty := true
+
+			for _, serverName := range serverNames {
+				serverTag := archive.TagServer(serverName)
+				streamDetails := &server.StreamDetail{}
+				err := r.Load(streamDetails, accountTag, streamTag, serverTag, typeTag)
+				if errors.Is(err, archive.ErrNoMatches) {
+					cmd.logWarning(
+						"Artifact not found: %s for stream %s in account %s by server %s",
+						typeTag.Value,
+						streamName,
+						accountName,
+						serverName,
+					)
+					continue
+				} else if err != nil {
+					return Skipped, fmt.Errorf("failed to lookup stream artifact: %w", err)
+				}
+
+				if streamDetails.State.LastSeq > 0 {
+					streamIsEmpty = false
+				}
+
+				replicasStreamDetails[serverName] = streamDetails
+				// Track streams with least one artifact
+				streamsInspected[accountName+"/"+streamName] = nil
+			}
+
+			// Check that all replicas are not too far behind the replica with the highest message & byte count
+			if !streamIsEmpty {
+				// Find the highest lastSeq
+				highestLastSeq, highestLastSeqServer := uint64(0), ""
+				for serverName, streamDetail := range replicasStreamDetails {
+					lastSeq := streamDetail.State.LastSeq
+					if lastSeq > highestLastSeq {
+						highestLastSeq = lastSeq
+						highestLastSeqServer = serverName
+					}
+				}
+				cmd.logDebug(
+					"Stream %s / %s highest last sequence: %d @ %s",
+					accountName,
+					streamName,
+					highestLastSeq,
+					highestLastSeqServer,
+				)
+
+				// Check if some server's sequence is below warning threshold
+				lastSequenceThreshold := uint64(math.Max(0, float64(highestLastSeq)-(float64(highestLastSeq)*checkLaggingStreamReplicasThreshold)))
+				for serverName, streamDetail := range replicasStreamDetails {
+					lastSeq := streamDetail.State.LastSeq
+					if lastSeq < lastSequenceThreshold {
+						examples.Addf(
+							"%s/%s server %s lastSequence: %d is behind highest lastSequence: %d on server: %s",
+							accountName,
+							streamName,
+							serverName,
+							lastSeq,
+							highestLastSeq,
+							highestLastSeqServer,
+						)
+						laggingReplicas += 1
+					}
+				}
+			}
+		}
+	}
+
+	cmd.logInfo("Inspected %d streams across %d accounts", len(streamsInspected), len(accountsWithStreams))
+
+	if laggingReplicas > 0 {
+		cmd.logIssue("Found %d replicas lagging behind", laggingReplicas)
+		cmd.logExamples(examples)
+		return SomeIssues, nil
+	}
+	return Pass, nil
+}
+
+const cpuUsageThreshold = 0.9 // Warn if any server is using more than 90% of the available CPU
```

Comment:
@jarretlavallee

Reply:
Is this a normalized %? The CPU usage reported from the endpoints maxes out at 100*cores.